### PR TITLE
Move ServiceCollection to abstractions

### PR DESCRIPTION
**Pull request by davidfowl**
Part of https://github.com/dotnet/runtime/issues/44728

~PS: I don't think I did this correctly~

### CODE REVIEW

File: src/libraries/Microsoft.Extensions.DependencyInjection/ref/Microsoft.Extensions.DependencyInjection.Forwards.cs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// ------------------------------------------------------------------------------
+// Changes to this file must follow the https://aka.ms/api-review process.
+// ------------------------------------------------------------------------------
+
+[assembly: System.Runtime.CompilerServices.TypeForwardedTo(typeof(Microsoft.Extensions.DependencyInjection.ServiceCollection))]
```
Comment (eerhardt) [inline src/libraries/Microsoft.Extensions.DependencyInjection/ref/Microsoft.Extensions.DependencyInjection.Forwards.cs:7]: You need a TypeForwardedTo in the `src` project as well.

Comment (davidfowl) [reply src/libraries/Microsoft.Extensions.DependencyInjection/ref/Microsoft.Extensions.DependencyInjection.Forwards.cs:7]: Ah

Comment (davidfowl) [reply src/libraries/Microsoft.Extensions.DependencyInjection/ref/Microsoft.Extensions.DependencyInjection.Forwards.cs:7]: Done

**Review by maryamariyan — APPROVED** [review]
LGTM

**Comment by maryamariyan** [comment]
Would any tests need to move too?

**Review by eerhardt — APPROVED** [review]

**Comment by davidfowl** [comment]
> Would any tests need to move too?

The tests are in a single project.

**Pull request merged**